controllers/machinehealthcheck_targets.go

```diff
@@ -116,16 +117,22 @@ func (t *healthCheckTarget) needsRemediation(logger logr.Logger, timeoutForMachi
 
 	// the node has not been set yet
 	if t.Node == nil {
-		// status not updated yet
-		if t.Machine.Status.LastUpdated == nil {
-			return false, timeoutForMachineToHaveNode
+		// TODO change to checking ControlPlaneReadyCondition in v1alpha4, when cluster.spec.controlPlaneRef will be required.
+		// We can't do this yet because ControlPlaneReadyCondition is only set when you're using a control plane provider,
+		// and that is optional in v1alpha3.
+		if !conditions.Has(t.Cluster, clusterv1.InfrastructureReadyCondition) || conditions.IsFalse(t.Cluster, clusterv1.InfrastructureReadyCondition) {
```

Comment:
If I got the problem right, this is accurate only for the first controlplane machine, but for all the other machines the actual startup time triggers when status.ControlPlaneInitialized==True (or ControlPlaneAvailable condition = True as suggested above).
However, I'm wondering if this assumption is somehow related to how kubeadm works and can't be used as a generic rule...

Reply:
Are you talking about KCP/MHC remediation?

In this PR's current form, given that MHC only remediates non-control plane Machines, this should be accurate for all non-control plane Machines.

Or am I misunderstanding what you're saying?

Reply:
How is this going to impact the remediation support for control plane nodes?

Reply:
I need to take that into account, since when we switched to using conditions for MHC, we got rid of the logic that only has MHC work on Machines owned by MachineSets and never control plane machines.